_ml-commons-plugin/gpu-acceleration.md

```diff
@@ -0,0 +1,285 @@
+
+#Install Neuron PyTorch
+pip install torch-neuron torchvision
+# If need to trace neuron model, install torch neuron with this command
+# pip install torch-neuron neuron-cc[tensorflow] "protobuf<4" torchvision
+
+# If need to trace neuron model, install transformers for tracing Huggingface model.
+# pip install transformers
+
+# Copy torch neuron lib to OpenSearch
+PYTORCH_NEURON_LIB_PATH=~/pytorch_venv/lib/python3.7/site-packages/torch_neuron/lib/
+mkdir -p $OS_HOME/lib/torch_neuron; cp -r $PYTORCH_NEURON_LIB_PATH/ $OS_HOME/lib/torch_neuron
+export PYTORCH_EXTRA_LIBRARY_PATH=$OS_HOME/lib/torch_neuron/lib/libtorchneuron.so
+echo "export PYTORCH_EXTRA_LIBRARY_PATH=$OS_HOME/lib/torch_neuron/lib/libtorchneuron.so" | tee -a ~/.bash_profile
+# Increase JVm stack size to >=2MB
+echo "-Xss2m" | tee -a $OS_HOME/config/jvm.options
+# Increase max file descriptors to 65535
+echo "$(whoami) - nofile 65535" | sudo tee -a /etc/security/limits.conf
+# max virtual memory areas vm.max_map_count to 262144
+sudo sysctl -w vm.max_map_count=262144
+```
+
+## (AWS Inferntia) Install GPU accelerator drivers manually
+
+If the previous two scripts do not provision your GPU accelerated node properly, you can install the drivers for AWS Inferentia manually.
+
+1. Deploy an AWS accelerator instance based on your chosen Linux operating system. For instructions, [Deploy on AWS accelerator instance](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/frameworks/torch/torch-neuron/setup/pytorch-install.html#deploy-on-aws-ml-accelerator-instance).
+
+2. (Optional) To monitor GPU usage of your accelerator instance, install [Neuron tools](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/frameworks/torch/torch-neuron/setup/pytorch-install.html), which allows models to be used inside your instance.
+
+  ```
+  # Install Neuron Tools
+  sudo apt-get install aws-neuronx-tools -y
+
+  export PATH=/opt/aws/neuron/bin:$PATH
+
+  # Test 
+  neuron-top
+  ```
+
+3. Copy the Neuron library into OpenSearch.
+
+  ```
+  OS_HOME=<OpenSearch installation path>
+  # For example, if you install OS_HOME in your home folder, it will be 
+  # OS_HOME=~/opensearch-2.5.0
+
+  # Activate pytorch_venv first if you haven't. Refer to "Install Driver" part
+  source pytorch_venv/bin/activate
+
+ 
+  # Set pytorch neuron lib path. In this example, we create pytorch_venv in home folder, so 
+  PYTORCH_NEURON_LIB_PATH=~/pytorch_venv/lib/python3.7/site-packages/torch_neuron/lib/
+
+
+  mkdir -p $OS_HOME/lib/torch_neuron; cp -r $PYTORCH_NEURON_LIB_PATH/ $OS_HOME/lib/torch_neuron
+  export PYTORCH_EXTRA_LIBRARY_PATH=$OS_HOME/lib/torch_neuron/lib/libtorchneuron.so
+  ```
+
+4. To make sure we have enough memory to upload a model, increase the JVM stack size to `>+2MB`. 
+
+  ```
+  echo "-Xss2m" | sudo tee -a $OS_HOME/config/jvm.options
+  ```
+
+5. Start OpenSearch. 
+
+### Troubleshooting
+
+If OpenSearch does not start, you'll receive a response similar to the following:
```

Comment:
There are a million reasons for why OpenSearch might not start - the max file descriptors and max memory maps are mandatory system settings so you should prescribe these somewhere above in the guide, otherwise you might as well rename this header to:

"Because OpenSearch won't launch with this configuration..."

Or if you feel that there are some instances where those system properties don't need to be set for a manual tar install, rephrase this section to something like: "If OpenSearch does not start, and the following log lines are present in the application logs, then..." or something along those lines.